src/crimson/os/seastore/cached_extent.h

```diff
@@ -284,7 +284,8 @@ class CachedExtent : public boost::intrusive_ref_counter<
   bool is_pending() const {
```

Comment:
How about splitting into the following two methods:

```c++
/// Returns true if extent can be mutated in an open transaction
bool is_mutable() const {
  return state == extent_state_t::INITIAL_WRITE_PENDING ||
      state == extent_state_t::MUTATION_PENDING ||
      state == extent_state_t::EXIST_MUTATION_PENDING;
}

/// Returns true if extent is part of an open transaction
bool is_pending() const {
  return is_mutable() || state == extent_state_t::EXIST_CLEAN;
}
```

The two methods imply different usage scenarios:
* Most of the existing users related to the `Cache::duplicate_for_write()` path should check `is_mutable()` for the decision.
* LBA-ptrs checks the new `is_pending()` to identify if it is part of a transaction.